### PR TITLE
limactl ls $INSTANCE should exit with non-0 status if $INSTANCE does not exist #1139 #2295

### PR DESCRIPTION
**Pull request by BajuMcBites**
- I believe this should fix the issue where nothing is printed on an unmatched list error caused by PR #2295
- Prints the matching instances, and then exits with 1
- Please let me know if I missed anything

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Please squash commits

File: cmd/limactl/list.go [hunk 1]
```diff
@@ -174,6 +179,17 @@ func listAction(cmd *cobra.Command, args []string) error {
...
+		if len(instanceNames) > 0 {
+			err := store.PrintInstances(out, instances, format, &options)
+			if err != nil {
+				return err
+			}
+		}
+		os.Exit(1)
+	}
+
 	return store.PrintInstances(out, instances, format, &options)
```
Comment (jandubois) [inline cmd/limactl/list.go:193]: This looks simpler (and avoids code duplication):

```suggestion
	err := store.PrintInstances(out, instances, format, &options)
	if err == nil && unmatchedInstance {
		os.Exit(1)
	}
	return err
```

I have also a slight preference to using the plural form `unmatchedInstances` over the singular, to match the "one or more" part.

**Review by jandubois — APPROVED** [review]
Thanks, LGTM. Any reason this is still in draft?

**Comment by BajuMcBites** [comment]
No, I was just waiting for the tests to run before I marked it as ready for review. Thanks

**Pull request merged**